### PR TITLE
Add type to array parameter in tool specification

### PR DESCRIPTION
**Pull request by Wadimz**
This PR introduces typed array parameters for tool specification, which is currently fails with some APIs with next error (OpenAI example):

> WARN: Exception was thrown on attempt 1 of 3: dev.ai4j.openai4j.OpenAiHttpException: {
>   "error": {
>     "message": "Invalid schema for function 'xxxx': In context=('properties', 'arg0'), array schema missing items",
>     "type": "invalid_request_error",
>     "param": null,
>     "code": null
>   }
> }

New test covers use case scenario that requires array parameter to be used for tool, and that was failing with error above prior the change.

### CODE REVIEW

File: langchain4j-core/src/main/java/dev/langchain4j/agent/tool/ToolSpecifications.java [hunk 1]
```diff
@@ -105,6 +101,52 @@ static Iterable<JsonSchemaProperty> toJsonSchemaProperties(Parameter parameter)
         return removeNulls(OBJECT, description); // TODO provide internals
     }
 
+    private static JsonSchemaProperty defineArrayItemTypeFromType(Type parameterizedType) {
```
Comment (langchain4j) [inline langchain4j-core/src/main/java/dev/langchain4j/agent/tool/ToolSpecifications.java]:
```suggestion
    private static JsonSchemaProperty arrayTypeFrom(Type type) {
```

File: langchain4j/src/test/java/dev/langchain4j/service/AiServicesIT.java [hunk 2]
```diff
@@ -1086,4 +1088,97 @@ void should_execute_multiple_tools_in_parallel_then_answer() {
...
+            return String.join("_", uuids);
+        }
+    }
+
+    @Test
+    void should_use_tool_with_array_parameters_then_answer() {
+        ToolSpecification listProcessorSpecification = ToolSpecification.builder()
+            .name("processUuids")
+            .description("processes list of uuids")
+            .addParameter("arg0", ARRAY, JsonSchemaProperty.items(JsonSchemaProperty.STRING), JsonSchemaProperty.description("uuids to use"))
```
Comment (langchain4j) [inline langchain4j/src/test/java/dev/langchain4j/service/AiServicesIT.java]:
```suggestion
            .addParameter("arg0", ARRAY, items(STRING), description("uuids to use"))
```

Please use static imports

File: langchain4j/src/test/java/dev/langchain4j/service/AiServicesIT.java [hunk 3]
```diff
@@ -1086,4 +1088,97 @@ void should_execute_multiple_tools_in_parallel_then_answer() {
...
+
+        @Tool("processes list of uuids")
+        String processUuids(@P("uuids to use") List<String> uuids) {
+            return String.join("_", uuids);
+        }
+    }
+
+    @Test
+    void should_use_tool_with_array_parameters_then_answer() {
+        ToolSpecification listProcessorSpecification = ToolSpecification.builder()
```
Comment (langchain4j) [inline langchain4j/src/test/java/dev/langchain4j/service/AiServicesIT.java]:
```suggestion
        ToolSpecification expectedToolSpecification = ToolSpecification.builder()
```

File: langchain4j-core/src/main/java/dev/langchain4j/agent/tool/ToolSpecifications.java [hunk 4]
```diff
@@ -73,29 +75,23 @@ static Iterable<JsonSchemaProperty> toJsonSchemaProperties(Parameter parameter)
...
 
-        if (type.isArray()
-                || type == List.class
-                || type == Set.class) { // TODO something else?
-            return removeNulls(ARRAY, description); // TODO provide type of array?
+        if (type.isArray()) {
+            return removeNulls(ARRAY, defineArrayItemTypeFromClass(type.getComponentType()), description);
+        }
+        if (Collection.class.isAssignableFrom(type)) {
+                return removeNulls(ARRAY, defineArrayItemTypeFromType(parameter.getParameterizedType()), description);
```
Comment (langchain4j) [inline langchain4j-core/src/main/java/dev/langchain4j/agent/tool/ToolSpecifications.java]:
```suggestion
            return removeNulls(ARRAY, defineArrayItemTypeFromType(parameter.getParameterizedType()), description);
```

File: langchain4j-core/src/main/java/dev/langchain4j/agent/tool/ToolSpecifications.java [hunk 5]
```diff
@@ -105,6 +101,52 @@ static Iterable<JsonSchemaProperty> toJsonSchemaProperties(Parameter parameter)
         return removeNulls(OBJECT, description); // TODO provide internals
     }
 
+    private static JsonSchemaProperty defineArrayItemTypeFromType(Type parameterizedType) {
+        if (parameterizedType instanceof ParameterizedType) {
+            ParameterizedType type = (ParameterizedType) parameterizedType;
```
Comment (langchain4j) [inline langchain4j-core/src/main/java/dev/langchain4j/agent/tool/ToolSpecifications.java]:
```suggestion
            ParameterizedType parameterizedType = (ParameterizedType) type;
```

**Review by ghost — COMMENTED** [review]
@Wadimz great job, thank you! I have left a few minor comments, please check

**Comment by Wadimz** [comment]
Thanks for the review! Updated according to comments. 
Added minor update to IT: `processes list of uuids` => `processes list of uuids into result string`
to make model response more predictable.

**Review by ghost — APPROVED** [review]
@Wadimz great job, thank you!

**Pull request merged**